connect-rest-extension/src/main/java/io/debezium/kcdrestextension/DebeziumResource.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.maven.artifact.versioning.ComparableVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.kcdrestextension.entities.TransformsInfo;
+
+@Path("/debezium")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class DebeziumResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebeziumResource.class);
+
+    // TODO: This should not be so long. However, due to potentially long rebalances that may have to wait a full
+    // session timeout to complete, during which we cannot serve some requests. Ideally we could reduce this, but
+    // we need to consider all possible scenarios this could fail. It might be ok to fail with a timeout in rare cases,
+    // but currently a worker simply leaving the group can take this long as well.
+    public static final long REQUEST_TIMEOUT_MS = 90 * 1000;
+    // Mutable for integration testing; otherwise, some tests would take at least REQUEST_TIMEOUT_MS
+    // to run
+    private static long requestTimeoutMs = REQUEST_TIMEOUT_MS;
+
+    private final List<TransformsInfo> transforms;
+    private Boolean isTopicCreationEnabled = null;
+    private final Herder herder;
+    private final Map<String, ?> config;
+
+    private static final ComparableVersion TOPIC_CREATION_KAFKA_VERSION = new ComparableVersion("2.6.0");
+
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public DebeziumResource(ConnectClusterState clusterState, Map<String, ?> config) {
+        Field herderField;
+        try {
+            herderField = ConnectClusterStateImpl.class.getDeclaredField("herder");
+        }
+        catch (NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+        herderField.setAccessible(true);
+        try {
+            this.herder = (Herder) herderField.get(clusterState);
+        }
+        catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        this.transforms = new ArrayList<>();
+        this.config = config;
+    }
+
+    // For testing purposes only
+    public static void setRequestTimeout(long requestTimeoutMs) {
+        DebeziumResource.requestTimeoutMs = requestTimeoutMs;
+    }
+
+    public static void resetRequestTimeout() {
+        DebeziumResource.requestTimeoutMs = REQUEST_TIMEOUT_MS;
+    }
+
+    @GET
+    @Path("/transforms")
+    public List<TransformsInfo> listTransforms() {
+        return this.getTransforms();
+    }
+
+    private synchronized List<TransformsInfo> getTransforms() {
+        if (this.transforms.isEmpty()) {
+            Iterator var1 = this.herder.plugins().transformations().iterator();
+
+            while (var1.hasNext()) {
+                PluginDesc<Transformation> plugin = (PluginDesc) var1.next();
+
+                if ("org.apache.kafka.connect.runtime.PredicatedTransformation".equals(plugin.className())) {
```

Comment:
Can we reference the actual class object and its name?

Reply:
Maybe... if someone uses a Kafka Connect distribution where the class is not included, that won't work. Like this I was able to avoid that dependency. wdyt?

Reply:
Ok; the usage of the class objects on the next lines is a bit odd then though. In all likelyhood any existing JVM will only load those classes when actually going into this branch, but I don't think there's a guarantee for it. But we can leave it as is and adjust if there actually is a problem.